### DISCUSSION TITLE
[Released] Support for Python 3.7+

### DISCUSSION THREAD

**Discussion by illvart** [vercel/community#55] [Idea]
[Update] Python 3.9 is available to all deployments today.

https://vercel.com/changelog/python-3-9-is-now-available

**Comment by styfle** [comment]
We are tracking this feature internally in Story 1908, thanks!

**Reply by illvart** [reply]
> We are tracking this feature internally in Story 1908, thanks!

This done so far?

**Reply by FalseDev** [reply]
> We are tracking this feature internally in Story 1908, thanks!

Are there any updates with this yet?

**Reply by wkpn** [reply]
@styfle any updates on that?

**Comment by rclement** [comment]
May I also suggest to support dynamic runtime specification please? Apparently this is already partially supported for Node.js (only major versions).

As Vercel is already supporting `pipenv` / `Pipfile` / `Pipfile.lock`, could it honor the specified Python version?
Here is the documentation: https://pipenv.pypa.io/en/latest/basics/#specifying-versions-of-python

I understand the underlying infrastructure is using AWS Lambda and I'm not sure what is possible, but ideally combined with `pyenv`, the process would be seamless, automatic and provide deterministic builds! (a first in serverless functions)

**Comment by whkoh** [comment]
Hi, any updates on supporting Python versions newer than 3.6.x?

**Comment by lslunis** [comment]
Any updates on this? Python 3.6 will reach end of life in two months: https://www.python.org/downloads/

**Comment by styfle** [comment]
~~You can try out Python 3.9 today by setting an environment variable `FORCE_RUNTIME_TAG=canary`.~~

~~If you're still using legacy [`builds`](https://vercel.com/docs/cli#project-configuration/builds), then you'll need to use `@vercel/python@canary`.~~

Python 3.9 is available to all deployments today 🎉 

https://vercel.com/changelog/python-3-9-is-now-available

**Reply by olivierphi** [reply]
Awesome! 😊
